packages/storybook/stories/InputDate.stories.tsx

```diff
@@ -0,0 +1,119 @@
+import type { Meta, StoryFn } from '@storybook/react'
+import {
+  DateInput,
+  StcStylesProvider,
+  DateInputProps,
+  Fieldset,
+} from '@stcland/components'
+import * as React from 'react'
+import customStyles from '../stc.config'
+
+const meta: Meta<typeof DateInput> = {
+  title: 'Components/Form/Input/DateInput',
+  component: DateInput,
+  argTypes: {
+    name: {
+      type: 'string',
+      value: 'dateInput',
+    },
+    value: {
+      type: 'string',
+      value: 1,
+    },
+    defaultValue: {
+      type: 'string',
+      value: '',
+    },
+    id: {
+      type: 'string',
+      value: 'number_id',
+    },
+    valueAsNumber: {
+      type: 'boolean',
+      value: false,
+    },
+    required: {
+      type: 'boolean',
+      value: false,
+    },
+    disabled: {
+      type: 'boolean',
+      value: false,
+    },
+    hidden: {
+      type: 'boolean',
+      value: false,
+    },
+    touched: {
+      type: 'boolean',
+      value: false,
+    },
+    dirty: {
+      type: 'boolean',
+      value: false,
+    },
+    onFocus: {
+      type: 'function',
+      value: (e: unknown) => console.log(e),
+    },
+    onChange: {
+      type: 'function',
+      value: (e: unknown) => console.log(e),
+    },
+  },
+}
+
+export default meta
+
+const DateInputTemplate: StoryFn<DateInputProps> = (args) => (
+  <Fieldset withPanel>
+    <DateInput {...args} />
+  </Fieldset>
+)
+
+export const Default = DateInputTemplate.bind({})
+Default.args = {
+  name: 'dateInput',
+  placeholder: 'DateInput',
+  label: 'DateInput Input',
+}
+
+export const Disabled = DateInputTemplate.bind({})
+Disabled.args = {
+  name: 'dateInput',
+  placeholder: 'DateInput',
+  label: 'DateInput Input',
+  disabled: true,
+}
+
+export const Required = DateInputTemplate.bind({})
+Required.args = {
+  name: 'dateInput',
+  placeholder: 'DateInput',
+  label: 'DateInput Input',
+  required: true,
+}
+
+export const Hidden = DateInputTemplate.bind({})
+Hidden.args = {
+  name: 'dateInput',
+  placeholder: 'DateInput',
+  label: 'DateInput Input',
+  hidden: true,
+}
+
+export const Touched = DateInputTemplate.bind({})
+Touched.args = {
+  name: 'dateInput',
+  placeholder: 'DateInput',
+  label: 'DateInput Input',
+  touched: true,
+}
```

Comment:
Same with `touched`? what is the expectative from a user perspective in the UI?

Reply:
touched and dirty are props injected by react-hook-form, usually its used to change state depending on the field. Since we dont have a design for it yet I havent implemented any change.